graf2d/gpad/src/TCanvas.cxx

```diff
@@ -1979,6 +1972,27 @@ void TCanvas::SetFolder(Bool_t isfolder)
    fgIsFolder = isfolder;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Set canvas name. In case `name` is an empty string, a default name is set.
+
+void TCanvas::SetName(const char *name)
+{
+   if (!name[0]) {
```

Comment:
Probably, one should extend check here and test:
```
if (!name || !name[0])
```
In both cases one expects that default name will be assigned

Reply:
I took that test from the existing code where the empty name was tested. Yes why not adding what you are suggesting.